src/interrupt.c

```diff
@@ -763,12 +766,12 @@ void disable_interrupts()
  * @brief Enable interrupts systemwide
  *
  * @note If this is called inside a nested disable call, it will have no effect on the
- *       system.  Therefore it is safe to nest disable/enable calls.  After the last
- *       nested interrupt is enabled, systemwide interrupts will be reenabled.
+ *       system.  Therefore it is safe to nest disable/enable calls.  After the least
+ *       nested enable call, systemwide interrupts will be reenabled.
  */
 void enable_interrupts()
 {
-    /* Don't do anything if we've hosed up or aren't initialized */
+    /* Don't do anything if we've hosed up (TODO: what does this mean?) or aren't initialized */
```

Comment:
I think the idea is "don't do anything if the interrupt depth is negative as all bets are off".

I think the comment and the if line is a relic from the past when libdragon didn't use assertions. Back at a time, if one called `enable_interrupts` at beginning of main by mistake (assuming it had to be done), it would forever run with interrupts disabled because of this and other similar lines. Now, it triggers an assertion instead, so this checks are not useful anymore (I think, but it should be easy to re-review this code and make sure what I wrote is correct).

In general, we switched approach and instead of silently doing nothing in bad situations, we now complain loudly. In fact, we consider a bug whenever libdragon freezes with no error messages.

Reply:
(dang this comment was written in 2012)

would
```suggestion
    /* Don't do anything if we aren't initialized */
```
be fine like the one in `disable_interrupts`, then
other comments indeed just refer to normal operation, not taking bugs into account